docs/Deprecating-Disabling-and-Removing-Formulae.md

```diff
@@ -0,0 +1,110 @@
+# Deprecating, Disabling, and Removing Formulae
+
+There are many reasons why formulae may be deprecated, disabled, or removed. This document explains the differences between each method as well as explaining when one method should be used over another.
+
+## Overview
+
+This general rule of thumb can be followed:
+
+- `deprecate!` should be used for formulae that _should_ no longer be used
+- `disable!` should be used for formulae that _cannot_ be used
+- Formulae that have been disabled for a long time or have a more serious issue should be removed
+
+## Deprecation
+
+If a user attempts to install a deprecated formula, they will be shown a warning message but the install will succeed.
+
+A formula should be deprecated to indicate to users that the formula should not be used and may be disabled in the future. Deprecated formulae should still be able to build from source and all bottles should continue to work. Users who choose to install deprecated formulae should not have any issues.
```

Comment:
I'm hesitant to use the word `maintained` here. If whether or not the project is maintained is our standard, when would a formula be deprecated rather than disabled? I think that deprecation _is_ for unmaintained formulae because it's saying "this will stop working in the future but for now it's okay". Then, when it does stop, it's disabled.

Reply:
> If whether or not the project is maintained is our standard

This is our docs so I think it's fair enough.

> I think that deprecation _is_ for unmaintained formulae because it's saying "this will stop working in the future but for now it's okay".

I think we're continuing to ensure that we don't break it in CI so we're still maintaining the formula (even if upstream are not maintaining the software). Agreed that may require a further rewording.

Reply:
Ah, so you're saying that the _formula_ is maintained by _us_, right? I agree with that.

How about, instead:

```
A formula should be deprecated to indicate to users that the formula should not be used and may be disabled in the future. Deprecated formulae should still build from source and their bottles should continue to work. These formulae should continue to receive the maintenance required to allow them to build.
```

Reply:
```suggestion
A formula should be deprecated to indicate to users that the formula should not be used and will be disabled in the future. Deprecated formulae should be maintained by the Homebrew maintainers so they can still build from source and their bottles continue to work (even if unmaintained upstream). If this is not possible, they should be disabled.
```

Reply:
I like that. Will do.